docs/components/Combobox/Combobox.stories.mdx

```diff
@@ -62,17 +84,17 @@ takes an explicit action to close it such as clicking outside the Combobox or
 pressing the ESC key.
 
 When `multiSelect` is enabled, a Combobox header will be displayed below the
-search input.
+search input. When no selections have been made, the header label will read
+"Select" and the button will read "Select all". When one or more options have
+been selected, the header label will show the number of selected items, like a
+counter. The button will then read "Clear".
```

Comment:
similar to other feedback, I wonder if we can be less specific and leverage the examples to show off what it looks like exactly. that way if we change the words, nothing or at least less, needs to be updated in the docs

so perhaps instead of describing each button state's wording, we say

".... a Combobox header will be displayed below the search input. This header provides details about the selection count, and a button to either select all or deselect all."

though to really get the most out of this abstracted approach we'd probably want to remove the specific references to the wording in the following paragraphs. I don't mind if we leave it like it.

Reply:
updated here 2c68e306ab2157618230bac7788580542b4848d2

I found a way to be less specific while still explaining the behaviour.

As far as examples, Combobox wasn't playing nicely in a Canvas so I've put that ticket in the backlog. For this section, I don't believe we need an example on the docs page (at least in this current structure of the docs) since we have a Multi Select story.